cmd/kpod/stop.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/storage/pkg/reexec"
+	"github.com/kubernetes-incubator/cri-o/server"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	"os"
+)
+
+const crioConfigPath = "/etc/crio/crio.conf"
+
+var stopTimeout int64
+
+var (
+	stopFlags = []cli.Flag{
+		cli.Int64Flag{
+			Name:        "timeout, t",
+			Usage:       "Seconds to wait for stop before killing it (default 10)",
+			Value:       10,
+			Destination: &stopTimeout,
+		},
+	}
+
+	stopDescription = "Stops one or more containers"
+	stopCommand     = cli.Command{
+		Name:        "stop",
+		Usage:       "Stop one or more containers",
+		Description: stopDescription,
+		Flags:       stopFlags,
+		Action:      stopCmd,
+		ArgsUsage:   "CONTAINER-ID [CONTAINER-ID...]",
+	}
+)
+
+func stopCmd(c *cli.Context) error {
+	var errorIDs []string
+	args := c.Args()
+	if len(args) < 1 {
+		return errors.Errorf("specify one or more container IDs")
+	}
+	config := new(server.Config)
+	if err := config.FromFile(crioConfigPath); err != nil {
+		if c.GlobalIsSet("config") || !os.IsNotExist(err) {
+			return err
+		}
+	}
+	if reexec.Init() {
+		return nil
+	}
+	service, err := server.New(config)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	for _, cid := range args {
+		r, err := service.StopContainer(context.Background(), &pb.StopContainerRequest{
+			ContainerId: cid,
+			Timeout:     stopTimeout,
+		})
+		if err != nil {
+			errorIDs = append(errorIDs, cid)
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+		}
+		logrus.Debugf("StopContainerResponse: %+v", r)
+	}
+	if len(errorIDs) == 0 {
+		return nil
+	}
+	// mimics docker stop behaviour; errors are put on stdout
+	// but return code needs to 1, so returning a blank error message
+	return fmt.Errorf("")
```

Comment:
We really dont want to exit as this is mimic'ing the behaviour of other container stop commands.  In this case, it always continues on to attempt others and then just exits with 1 at the end.